### PR TITLE
Logout clearing sessions and canceling pending api requests

### PR DESCRIPTION
**Pull request by mateuszmackowiak**

### CODE REVIEW

File: AirCasting/AuthViews/UserAuthenticationSession.swift [hunk 1]
```diff
@@ -29,8 +29,11 @@ final class UserAuthenticationSession: ObservableObject {
     }
 
     func deauthorize() throws {
-        self.token = nil
+        URLSession.shared.getAllTasks { tasks in
+            tasks.forEach { $0.cancel() }
+        }
```
Comment (smialko) [inline AirCasting/AuthViews/UserAuthenticationSession.swift:34]: You've already done that in the logout function, is there a reason to repeat canceling all tasks here?

**Review by smialko — COMMENTED** [review]
It's still possible to sign in with incorrect data. Is it something we can fix or it's up to the backend part?

File: AirCasting/AuthViews/UserAuthenticationSession.swift (hunk 1 above)
Comment (mateuszmackowiak) [reply AirCasting/AuthViews/UserAuthenticationSession.swift:34]: If there would be a pending call to api that synchronises the sessions there is no check if the user is logged in or not. so if we wouldn't cancel it the returned sessions would again be added into the database.

File: AirCasting/CoreData/SessionStorage.swift [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+final class SessionStorage: ObservableObject {
+    let persistenceController: PersistenceController
+    init(persistenceController: PersistenceController) {
+        self.persistenceController = persistenceController
+    }
+
+    func clearAllSession() throws {
+        let deleteRequest = NSBatchDeleteRequest(fetchRequest: SessionEntity.fetchRequest())
+        deleteRequest.resultType = .resultTypeObjectIDs
+        let res = try persistenceController.editContext().execute(deleteRequest)
```
Comment (pawelgil) [inline AirCasting/CoreData/SessionStorage.swift:16]: This code will not inform any `NSManagedObjectContext` instance in use about the change made to the store. Take a look at the [Apple guide](https://developer.apple.com/library/archive/featuredarticles/CoreData_Batch_Guide/BatchDeletes/BatchDeletes.html) to batch delete requests. IFK if this is the expected behavior here, just pointing this CoreData "gotcha" out.

Comment (mateuszmackowiak) [reply AirCasting/CoreData/SessionStorage.swift:16]: good catch. in current usage it was intended but maybe not a good idea in global scope.

**Review by smialko — APPROVED** [review]

**Pull request merged**